### DISCUSSION TITLE
New Sequent 4 relay/4 input now has pulse counters

### DISCUSSION THREAD

**Discussion by tagyoureit** [tagyoureit/nodejs-poolController#644] [Nixie]
https://sequentmicrosystems.com/products/four-relays-four-inputs-for-raspberry-pi

> The 4.0 release of the card replaced the I/O expander with a local processor. This enhancement allowed us to add several new features:
> * Independent RS485 (no Raspberry Pi needed)
> * MODBUS protocol allows the card to be controlled from any PLC
> * Transition counters to 400 Hz on all inputs
> * Input counters with cumulative or PPS (pulse per second) reading up to 400 Hz
> * 2 x Quadrature encoder inputs reading up to 400 PPS
> * PWM inputs on each channel, reading both the fill factor and the frequency

Who will be the first to try hooking up a flow meter to this?

**Comment by Telemachus1977** [comment]
The price didn't go up, either.

I feel like a DAC would be much more useful that a pulse counter.  That would be very close to an all-in-one easy-touch 4 replacement.

How does RS-485 work without a raspberry pi?

**Reply by tagyoureit** [reply]
Various people have wanted to use pulse counters to measure flow (instantaneous or total).  They come in all sizes but get very expensive if you want to measure flow on a 3" or 4" line.  There are cheaper ones for 1/2", 3/4", etc (check out the [Atlas flow meters](https://atlas-scientific.com/flow-meters)) and some folks would like to measure how much water is being added to the pool via the fill line.  The pulse counters wouldn't be used for measuring temps.

> How does RS-485 work without a raspberry pi?

Below I pasted the full text from the email announcement... it's more for modbus remote control of the card rather than being able to read/write generic rs485 like we are doing with the njsPC app.  


```
MAKING THE BEST OF SEMICONDUCTOR SHORTAGES

The 4-RELAY-4-INPUT card was part of our "dumb" cards set, using an I/O expander to read the inputs and set the outputs. Basic functionality was OK, adding an RS485 driver to allow Raspberry Pi to control the card using either I2C or MODBUS protocols. All was good until the I/O expanders started running out of stock. We changed the design using a 20 pin processor which seems to be in ample supply.

The REV4 of the 4-RELAY-4-INPUT card is no longer "dumb". In adition to reading AC or DC signals from 3V to 250V, and controlling loads up to 8A and 240VAC, the card has some interesting new features:

1. The RS485 transceiver can be driven by the local processor. The card can be controlled standalone or daisy-chained from a standard PLC using MODBUS. NO RASPBERRY PI NEEDED!

2. Inputs can be used as pulse counters, up to 400 Hz.

3. Pulses can be read cumulative or PPS (pulse per second).

4. Inputs can read frequency and fill factor of PWM signals, up to 100 Hz with 1% and up to 500 Hz with 5% precision.

5. Inputs can be paired to read two quadrature encoders up to 400Hz.
```

**Reply by Telemachus1977** [reply]
As this is the only relay board that sequent offers that has both NO and NC contacts at 8A, with RS485, it is still my preferred solution with your awesome software.  I have helped a few friends set up their systems using this card and a cheap ADC for inputs.

With the microcontroller chip they have added, there is so much more potential to this board.  Come on Sequent, take the next step!  Add 1K or 10K thermistors and
0-10V analog inputs and this is THE pool automation HAT.

**Comment by arieltools** [comment]
can this be controlled through an ethernet to rs485 adapter like 

https://www.amazon.com/gp/product/B0BJB2YWZF/ref=ppx_yo_dt_b_asin_title_o08_s00?ie=UTF8&psc=1

when standalone? (i.e.: not connectedd to a Pi but using its own processor)

**Reply by rstrouse** [reply]
The Sequent relay is an I2c device so it requires an MCU (like an ESP/Arduino) or CPU (like Raspberry Pi) to control it.  However this version has an MCU embedded but I don't know the capabilities.  You might need to brush up on some modbus skills to make it talk though.  Either way it would need some coding to get it to work.

The device you linked to is commonly used as a TCP interface to RS485 but that still requires raw protocol to be written to the socket.

**Reply by arieltools** [reply]
Thanks so what would you recommend as a solution for relays that control valves without having the CPU directly on the pad, but remote?

**Reply by rstrouse** [reply]
The way REM is written is so it manages the valve and circuit control with latches so if anything goes awry the latches would expire and the circuits/valves return to their off/idle state.  I have built a few of these and all of them with a single pi4 sitting out at the pad with local relay control.

However, that doesn't mean there are not other options.   In the end you will need some sort of controller at the pad.  This can simply be a processor running REM only.  There are warts with the Pi Zero 1 (I think it falls apart when you try to control more than one I2c device) but GPIO triggers seem to work fine.  If they ever start shipping the Pi Zero 2W again that may be a good alternative to an in between device.

I have thought about building an ESP32 based REM controller for the pad but then you have all of the external power requirements to deal with anyway.  The relays need trigger voltage and the valves need power so you are back at having the control equipment local anyway.

I am curious though.  Why not a processor on the pad?

**Reply by arieltools** [reply]
The way you've thought this off it does make sense to me to have a CPU on the pad just for REM and it's probably the route I'll end up going.

The main reason for no-CPU-on-the-pad was to have less software-nodes on the installation and just leave it to the central CPU to control it all using HA. Less stuff to maintain and easier to do backups.

One thing I want to achieve is to be able to [fully control the valve actuators](https://www.troublefreepool.com/threads/finally-installed-home-made-automation-to-include-variable-position-valve-control.237974/) - i.e: put the valves in any direction. Using async nodes on [NodeRed, mimicking "inching"](https://flows.nodered.org/flow/50ac7713a4373bb22a6d931ab0e9f7ee) this would be a simple task that I wanted to try.

Currently I'm able to control the Intelliflo pump using an [RS485-to-Ethernet adapter](https://www.amazon.com/gp/product/B0BJB2YWZF/ref=ppx_yo_dt_b_asin_title_o08_s00?ie=UTF8&psc=1) through Lan from my HA CPU. That's why I was thinking to simply control it all, including Intermatic valves, from the Lan, using the same RS485 bus (can introduce noise / not recommended) or a new one by simply adding another RS485-ETH adapter.

I like the sparsity of a system like this and wanted to try it's feasibility, albeit understanding [the recommended Nixie method](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#automated-valves) is to have it all on one Pi with HATs and/or a single Arduino board.
